app/src/main/java/dev/animeshvarma/sigil/ui/screens/LockScreen.kt

```
@@ -21,42 +21,65 @@ import androidx.compose.ui.Modifier
+ *
+ * This composable displays either a PIN/password entry UI or a biometric unlock button depending on
+ * the configured lock mode and runtime state, verifies PIN via the provided ViewModel,
+ * triggers biometric prompts (including automatic prompt on lifecycle resume when applicable),
+ * and shows dialogs for biometric invalidation and emergency reset. On successful authentication
+ * it invokes the provided unlock callback.
+ *
+ * @param viewModel The SigilViewModel used to read preferences, verify the app secret, and perform data wipe.
+ * @param onUnlock Callback invoked when authentication succeeds (PIN verified or biometric success).
+ */
 @Composable
 fun LockScreen(
```

detekt (1.23.8)

[warning] The function LockScreen appears to be too complex based on Cyclomatic Complexity (complexity: 32). Defined complexity threshold for methods is set to '25'

(detekt.complexity.CyclomaticComplexMethod)